common/transport/http/src/http.ts

```diff
@@ -26,6 +26,9 @@ export type HttpCallback = (err: Error, body?: string, response?: IncomingMessag
  *                  You generally want to use these higher-level objects (such as [azure-iot-device-http.Http]{@link module:azure-iot-device-http.Http}) rather than this one.
  */
 export class Http {
+
+  private _request: (options: RequestOptions | string, callback?: (res: IncomingMessage) => void) => ClientRequest = request;
```

Comment:
>_request [](start = 10, length = 8)

if this is used only for mocking in tests, why not use sinon stubs? plus the type/default value here is really hard to read and understand